tests/units/test_copier.py

```diff
@@ -0,0 +1,102 @@
+# Author: Felix Fontein <felix@fontein.de>
+# GNU General Public License v3.0+ (see LICENSES/GPL-3.0-or-later.txt or https://www.gnu.org/licenses/gpl-3.0.txt)
+# SPDX-License-Identifier: GPL-3.0-or-later
+# SPDX-FileCopyrightText: 2020, Ansible Project
+
+"""
+Test utils module.
+"""
+
+from __future__ import annotations
+
+import pathlib
+import subprocess
+from unittest import mock
+
+import pytest
+
+from antsibull_changelog.copier import GitCopier
+
+
+def test_git_copier(tmp_path_factory):
+    directory: pathlib.Path = tmp_path_factory.mktemp("changelog-test")
+
+    src_dir = directory / "src"
+    src_dir.mkdir()
+    (src_dir / "empty").touch()
+    (src_dir / "link").symlink_to("empty")
+    (src_dir / "dir").mkdir()
+    with (src_dir / "file").open("w", encoding="utf-8") as f:
+        f.write("content")
+    with (src_dir / "dir" / "binary_file").open("wb") as f:
+        f.write(b"\x00\x01\x02")
+
+    copier = GitCopier()
+
+    def assert_same(a: pathlib.Path, b: pathlib.Path):
+        if a.is_file():
+            assert b.is_file()
+            with a.open("rb") as f:
+                ac = f.read()
+            with b.open("rb") as f:
+                bc = f.read()
```

Comment:
You can use `pathlib.Path`'s `read_text()` method instead.

Reply:
I think `read_bytes()` is more appropriate :)

Reply:
05b32f18a3cb9d72ad8acada809eaf183ca458e0